src/commands/functions-delete.ts

```diff
@@ -0,0 +1,88 @@
+"use strict";
+
+import * as _ from "lodash";
+
+import { Command } from "../command";
+import * as clc from "cli-color";
+import * as cloudfunctions from "../gcp/cloudfunctions";
+import * as functionsConfig from "../functionsConfig";
+import { deleteFunctions } from "../functionsDelete";
+import * as getProjectId from "../getProjectId";
+import * as helper from "../functionsDeployHelper";
+import { prompt } from "../prompt";
+import { requirePermissions } from "../requirePermissions";
+import * as utils from "../utils";
+
+module.exports = new Command("functions:delete [filters...]")
+  .description("delete one or more Cloud Functions by name or group name.")
+  .option(
+    "--region <region>",
+    "Specify region of the function to be deleted. " +
+      "If omitted, functions from all regions whose names match the filters will be deleted. "
+  )
+  .option("-f, --force", "No confirmation. Otherwise, a confirmation prompt will appear.")
+  .before(requirePermissions, ["cloudfunctions.functions.list", "cloudfunctions.functions.delete"])
+  .action(async (filters, options) => {
+    if (!filters.length) {
+      return utils.reject("Must supply at least function or group name.");
+    }
+
+    const projectId = getProjectId(options);
+
+    // Dot notation can be used to indicate function inside of a group
+    const filterChunks = filters.map((filter: string) => {
+      return filter.split(".");
+    });
+    const config = await functionsConfig.getFirebaseConfig(options);
+    const appEngineLocation = functionsConfig.getAppEngineLocation(config);
+    const existingFns = await cloudfunctions.listAllFunctions(projectId);
+    const functionsToDelete = existingFns.filter((fn) => {
+      const regionMatches = options.region ? helper.getRegion(fn.name) === options.region : true;
```

Comment:
nit - this might be a simpler way to express this:
!options.region || helper.getRegion(fn.name) === options.region

It's only a bit shorter, but to me it's more comprehensible because I don't have to first mentally convert it to an if statement.